packages/evm/src/fungibleAssetTransfer.ts

```diff
@@ -95,71 +18,99 @@ export async function createEvmFungibleAssetTransfer(
  * for transferring fungible tokens
  * using Sygma protocol
  */
-class EvmFungibleAssetTransfer extends EvmTransfer {
-  protected destinationAddress: string = '';
+class FungibleAssetTransfer extends AssetTransfer {
+  /**
+   * @privateRemarks
+   *
+   * will be used in future
+   */
   protected securityModel: SecurityModel;
-  protected adjustedAmount: bigint = BigInt(0);
-  private specifiedAmount: bigint; // Original value to transfer without deductions
+  /**
+   * @privateRemarks
+   *
+   * adjustedAmount is the amount that will
+   * be deducted from the users wallet
+   * specifiedAmount is the amount that the
+   * user wants to transfer
+   */
+  protected declare adjustedAmount: bigint;
+  protected specifiedAmount: bigint;
 
-  constructor(transfer: EvmFungibleTransferRequest, config: Config) {
+  /**
+   * Returns amount to be transferred considering the fee
+   */
+  get transferAmount(): bigint {
+    return this.adjustedAmount;
+  }
+
+  constructor(transfer: FungibleTransferParams, config: Config) {
     super(transfer, config);
     this.specifiedAmount = transfer.amount;
-
-    if (isValidAddressForNetwork(transfer.destinationAddress, this.destination.type))
-      this.destinationAddress = transfer.destinationAddress;
     this.securityModel = transfer.securityModel ?? SecurityModel.MPC;
   }
 
   /**
-   * Returns amount to be transferred considering the fee
+   * Returns encoded deposit
+   * data
+   * @returns {string}
    */
-  get amount(): bigint {
-    return this.adjustedAmount;
+  protected getDepositData(): string {
+    return createERCDepositData(this.adjustedAmount, this.recipient, this.destination.parachainId);
   }
 
-  public getSourceNetworkProvider(): Eip1193Provider {
-    return this.sourceNetworkProvider;
-  }
+  /**
+   * Checks the source wallet
+   * balance to see if transfer
+   * would fail
+   * @param {EvmFee} fee
+   * @returns {Promise<boolean>}
+   */
+  protected async hasEnoughBalance(fee?: EvmFee): Promise<boolean> {
+    const resource = this.resource as EvmResource;
+    const provider = new Web3Provider(this.sourceNetworkProvider);
 
-  async isValidTransfer(): Promise<boolean> {
-    const sourceDomainConfig = this.config.getDomainConfig(this.source) as EthereumConfig;
-    const web3Provider = new Web3Provider(this.sourceNetworkProvider);
-    const bridge = Bridge__factory.connect(sourceDomainConfig.bridge, web3Provider);
-    const { resourceId } = this.resource;
-    const handlerAddress = await bridge._resourceIDToHandlerAddress(resourceId);
-    return utils.isAddress(handlerAddress) && handlerAddress !== constants.AddressZero;
-  }
+    if (fee) {
+      switch (fee.type) {
+        case FeeHandlerType.BASIC:
```

Comment:
is it not implemented yet or we don't expect a Basic type?

Reply:
falls through and uses the same logic as `TWAP`